### PR TITLE
[ARTS-689][Subfeature] Practitioner use site service client

### PR DESCRIPTION
**Pull request by anatbal**
## Description
- Practitioner uses the new site-service-client

### Changes
- [ARTS-689] - Practitioner uses the new site-service-client
### Checklist:

- [x] Branch name follows convention (feature/arts-#-short-name OR fix/arts-#-short-name)
- [x] I have included a short-meaningful title, description and changes for this PR

[ARTS-689]: https://ndph-arts.atlassian.net/browse/ARTS-689

### CODE REVIEW

**Review by LizaShak — APPROVED** [review]

**Review by shajidissa — DISMISSED** [review]

File: practitioner-service/src/main/java/uk/ac/ox/ndph/mts/practitioner_service/validation/RoleAssignmentValidation.java [hunk 1]
```diff
@@ -35,12 +39,13 @@ public ValidationResponse validate(RoleAssignment entity) {
             return new ValidationResponse(false, "roleId must have a value");
         }
 
-        if (!this.roleServiceClient.entityIdExists(entity.getRoleId(), roleServiceClient.noAuth())) {
+        if (!this.roleServiceClient.entityIdExists(entity.getRoleId(), RoleServiceClient.noAuth())) {
```
Comment (darrengsmithson) [inline practitioner-service/src/main/java/uk/ac/ox/ndph/mts/practitioner_service/validation/RoleAssignmentValidation.java:42]: not sure about this..seems odd that we call a class and pass in that class as a param.... couldnt we pass a flag to state whether to bypass auth or check if the headers are null in RoleServiceClient and if so use noAuth?

**Review by darrengsmithson — CHANGES_REQUESTED** [review]

File: practitioner-service/src/main/java/uk/ac/ox/ndph/mts/practitioner_service/validation/RoleAssignmentValidation.java (hunk 1 above)
Comment (anatbal) [reply practitioner-service/src/main/java/uk/ac/ox/ndph/mts/practitioner_service/validation/RoleAssignmentValidation.java:42]: The thing here is that is only a temp solution, @LizaShak is already working on the role-service auth and will be replaced soon with actual authHeaders. WDYT?

Comment (darrengsmithson) [reply practitioner-service/src/main/java/uk/ac/ox/ndph/mts/practitioner_service/validation/RoleAssignmentValidation.java:42]: fair enough if its temp - can you add a comment?

Comment (anatbal) [reply practitioner-service/src/main/java/uk/ac/ox/ndph/mts/practitioner_service/validation/RoleAssignmentValidation.java:43]: Done

**Review by darrengsmithson — APPROVED** [review]

**Review by LizaShak — APPROVED** [review]

**Pull request merged**